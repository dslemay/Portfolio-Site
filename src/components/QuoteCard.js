import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import themeUtils from './themeUtils';
import Card from './Card';

const QuoteContainer = styled.div`
  position: relative;
  padding: 1rem 1rem 0rem 2rem;

  &::before {
    content: '\\201C';
    font-family: Georgia;
    font-style: bold;
    font-size: 4rem;
    color: ${themeUtils.complementaryDark};
    position: absolute;
    top: -1.4rem;
    left: 0.1rem;
  }

  &::after {
    content: '\\201D';
    font-family: Georgia;
    font-style: bold;
    font-size: 4rem;
    color: ${themeUtils.complementaryDark};
    position: absolute;
    bottom: -3.6rem;
    right: 0.1rem;
  }
`;

const QuoteCard = props => (
  <Card>
    <QuoteContainer>{props.children}</QuoteContainer>
  </Card>
);

QuoteCard.propTypes = {
  children: PropTypes.node.isRequired
};

export default QuoteCard;
