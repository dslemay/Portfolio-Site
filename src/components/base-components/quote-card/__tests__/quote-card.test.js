import React from 'react'
import { render } from 'react-testing-library'

import QuoteCard from '..'

describe('<QuoteCard />', () => {
  it('renders correctly', () => {
    const { container } = render(
      <QuoteCard>
        <p>Inspiring quotes are amazing!</p>
      </QuoteCard>,
    )
    expect(container).toMatchSnapshot()
  })
})
