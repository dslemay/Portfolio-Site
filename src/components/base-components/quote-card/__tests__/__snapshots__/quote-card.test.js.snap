// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuoteCard /> renders correctly 1`] = `
.emotion-1 {
  margin-bottom: 1.5rem;
  border: 1px solid #ccc;
  padding: 0.5rem 0.25rem;
  box-shadow: 3px 3px 5px #656E9A;
}

.emotion-0 {
  position: relative;
  padding: 1rem 1rem 0rem 2rem;
}

.emotion-0::before {
  content: '\\201C';
  font-family: Georgia;
  font-style: bold;
  font-size: 4rem;
  color: #9A8956;
  position: absolute;
  top: -1.4rem;
  left: 0.1rem;
}

.emotion-0::after {
  content: '\\201D';
  font-family: Georgia;
  font-style: bold;
  font-size: 4rem;
  color: #9A8956;
  position: absolute;
  bottom: -3.6rem;
  right: 0.1rem;
}

<div>
  <div
    class=" emotion-1"
  >
    <div
      class="emotion-0"
    >
      <p>
        Inspiring quotes are amazing!
      </p>
    </div>
  </div>
</div>
`;
